### DISCUSSION TITLE
[book] quarto does not copy SVG files included with <object> in the _book directory.

### DISCUSSION THREAD

**Discussion by ejeandel** [quarto-dev/quarto-cli#2710] [Q&A]
Hello
The .qmd of my book contain several .svg that are included with <object>.

To be exact, I'm using a simple extension I wrote that replace `{{< pic "toto.svg" >}}` by
`<object data="toto.svg" ...></object>`: 

```
function pic(args)
   local var1 = pandoc.utils.stringify(args[1])
   if FORMAT == "html" then
      return pandoc.RawInline('html', '<object data="'.. var1 .. " width="100%" type="image/svg+xml"></object>' )
   else ...
   end
end
```

I'm not sure if the fact I'm using an extension is relevant, but that's just in case.

When I type quarto preview --html, the svg files are not copied to the _book directory, and therefore are not accessible in the preview.

Can I change something in the lua file to force quarto to copy the file ? or list somewhere in the yaml the files that should be copied ?

Thanks, 

E.

**Comment by cscheid** [comment]
You can directly add a file to be included with

```
quarto.doc.add_format_resource(filename)
```

Note that this feature requires a recent, prerelease version of quarto: your version string should say 1.2.*

**Reply by ejeandel** [reply]
Thanks! Can I expect this feature to come in the 1.1.x version of quarto at some point ? Is there a reason for me to stick with the 1.1.x version rather than using the prerelease version 1.2.x ?

**Reply by cscheid** [reply]
If you can install software on your machine, I would recommend switching to the prereleases; quarto is still under very active development, so you'll get a number of new features: https://github.com/quarto-dev/quarto-cli/blob/main/news/changelog-1.2.md
